website/docs/sql/datatype/array.md

```diff
@@ -0,0 +1,93 @@
+
+Generally, array values are denoted in curly braces with elements separated by commas, like so:
+
+```
+{"Lorem", "ipsum", "dolor", "sit", "amed"}
+```
+
+## Element Types and Nullability
+
+Hyper's arrays are _strongly typed_:
+All elements must be of the same type – the array's _element type_.
+The element type is a defining part of the array's overall type, meaning that `array(integer)` is a different type than `array(boolean)` or `array(text)`.
+Arrays can be built from all [atomic types](./index.md) available in Hyper.
+
+Part of an array's element type is the element's nullability. 
+For example, `array(smallint)` is different from `array(smallint not null)`.
+Note that this is independent of the nullability of the array itself.
+The following four options all represent different types in Hyper:
+
+|Type|array nullable?|elements nullable?| possible values|
+|---|---|---|---|
+|`array(integer)`|✅|✅|`{}`,`{1,2,3}`,`{1,2,null}`, `null`|
+|`array(integer not null)`|✅|❌|`{}`,`{1,2,3}`,`null`|
+|`array(integer) not null`|❌|✅|`{}`,`{1,2,3}`,`{1,2,null}`|
+|`array(integer not null) not null`|❌|❌|`{}`,`{1,2,3}`|
+
+The inner nullability of an array type can be changed by casting, using the conventional [cast syntax](../scalar_func/conversion.md):
+
+```sql
+# nullable to non-nullable
+> select ('{1,2,3}'::array(integer))::array(integer not null)
+# non-nullable to nullable
+> select ('{1,2,3}'::array(integer not null))::array(integer)
+```
+
+A cast from a non-nullable element type to its nullable counterpart always succeeds.
+The reverse direction (nullable to non-nullable) succeeds if the array does not contain `null` elements; otherwise the cast results in an error.
+Casts across element types (e.g. from `array(integer not null)` to `array(bigint not null)`) are currently not supported.
+
+:::info
+Non-nullable element types use less memory and enable optimizations for certain array operations. Users are therefore advised to use the most "restrictive" element type possible, given the use case at hand. 
+:::
+
+
+The length (i.e., the number of elements) of an array in Hyper is not part of its type.
+While arrays inside a column must have the same element type, they can be of different length.
+
+## Working with Arrays
+
+Arrays can be created in two ways:
+
+ - Using the type constructor syntax:
+   ```sql
+   > select array[1,2,3];
+   {1,2,3}
+   ```
+ - Using casts from string data:
+   ```sql
+   > select '{1,2,3}'::array(integer);
+   {1,2,3}
+   ```
+
+Note that using the constructor syntax, the array type will be inferred automatically.
+If the constructed array does not contain a `null` value, the element type will be inferred as non-nullable.
+
+
+Array elements can be retrieved using the conventional bracket-indexing notation. Indexes start at one.
+```sql
+> select ('{1,1,2,3,5}'::integer[])[4]
+3
+```
+
+:::info
+Arrays are one-dimensional. Higher-dimensional objects (e.g., matrices) must be flattened explicitly.
+:::
```

Comment:
This info box doesn't seem to work. At least it doesn't work in[ the preview on git-hub](https://github.com/tableau/hyper-db/blob/lwolf-W-15542353-document-array-type/website/docs/sql/datatype/array.md#:~:text=Non%2Dnullable%20element%20types%20use%20less%20memory%20and%20enable).

Reply:
<img width="1031" alt="Screenshot 2024-08-07 at 09 53 38" src="https://github.com/user-attachments/assets/49607e99-e0cb-4fd3-8a6d-878205f50162">